### PR TITLE
Fix for issue #582 - update-properties does not work across parent-child pom 

### LINKED ISSUE #582: update-properties does not work across parent-child pom
Version 2.8.1 and above

Parent POM has

```
<properties>
  <version.lib>1.0.0</version.lib>
</properties>

  <dependencies>
    <dependency>
      <groupId>org.example</groupId>
      <artifactId>someLib</artifactId>
      <version>${version.lib}</version>
    </dependency>
  <dependencies>
```
Child POM has

```
<properties>
  <version.lib>1.1.0</version.lib>
</properties>
```
The child $version.lib property is active (i.e. does control the version of someLib in use) however update-properties does not update $version.lib to latest version when run on the child POM. They will do this if the dependency on someLib exists in the child pom.

### PR DESCRIPTION
**Pull request by prodj17**
Fixed #582
 o Updated PomHelper to validate version properties also across parents.

### CODE REVIEW

**Comment by prodj17** [comment]
@slawekjaranowski do you think you can find some time to look into this?

File: src/it/it-update-properties-issue-582/pom.xml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    <version>1.0</version>
+    <packaging>pom</packaging>
+    <name>update-properties with one property, with dependency in parent</name>
+
+    <properties>
+        <api>1.0</api>
+    </properties>
+
+    <build>
+        <pluginManagement>
```
Comment (slawekjaranowski) [inline src/it/it-update-properties-issue-582/pom.xml:21]: the same - do we need this?

File: src/it/it-update-properties-issue-582/verify.bsh [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import java.io.*;
```
Comment (slawekjaranowski) [inline src/it/it-update-properties-issue-582/verify.bsh:1]: please use goovy  script

**Review by slawekjaranowski — COMMENTED** [review]
- it looks like we can add a unit tests for PomHelper, it should be easier to check coverage by test
- what will be if in parent we will have dependencyManagment and in child dependency without version, and property
- what will be if in parent we will have dependencyManagment and in child no dependency, but only property

disclosure - I'm not know this code ... and logic looks quite complex, so will be good if someone else looks at it

**Comment by prodj17** [comment]
> disclosure - I'm not know this code ... and logic looks quite complex, so will be good if someone else looks at it

The change in logic is actually quite simple, yet for some reason github diff messed up. 

The logic of dependencyManagement vs property stays the same as for a single pom. The only change there is to actually iterate over the parent poms and process those as well when looking for properties.

@olamy @bmarwell I've seen you made changes to PomHelper more recently. Do you guys think you can look into it?

File: src/it-repo/dummy-parent-issue-582-1.0.pom [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+  <dependencies>
+    <dependency>
+      <groupId>localhost</groupId>
+      <artifactId>dummy-api</artifactId>
+      <version>${api}</version>
+    </dependency>
+  </dependencies>
+
+  <build>
+    <pluginManagement>
```
[the remark this answers is not in the document]
Comment (prodj17) [reply src/it-repo/dummy-parent-issue-582-1.0.pom:23]: Not really. I will remove

File: src/it/it-update-properties-issue-582/pom.xml (hunk 1 above)
Comment (prodj17) [reply src/it/it-update-properties-issue-582/pom.xml:21]: same. will remove

File: src/it/it-update-properties-issue-582/verify.bsh (hunk 2 above)
Comment (prodj17) [reply src/it/it-update-properties-issue-582/verify.bsh:1]: ok

**Comment by slawekjaranowski** [comment]
please change commit message to something more related what we do, eg like:

```
update-properties does not work across parent-child pom
```

pleas also squash to one commit

**Comment by prodj17** [comment]
> please change commit message to something more related what we do, eg like:
> 
> ```
> update-properties does not work across parent-child pom
> ```
> 
> pleas also squash to one commit

@slawekjaranowski done

**Comment by prodj17** [comment]
@slawekjaranowski It has been more than a week and there are no comments. Could we merge this PR?

**Comment by slawekjaranowski** [comment]
Ok.
last request, can you squash to single commit. 
Of course I can do it for you.

**Comment by slawekjaranowski** [comment]
@prodj17 - thanks

**Pull request merged**